src/components/content-tab-assistant.tsx

```diff
@@ -335,15 +336,29 @@ export function ContentTabAssistant( { selectedSite }: ContentTabAssistantProps
 	const { userCanSendMessage } = usePromptUsage();
 	const { fetchAssistant, isLoading: isAssistantThinking } = useAssistantApi( selectedSite.id );
 	const { messages: welcomeMessages, examplePrompts } = useWelcomeMessages();
-	const [ input, setInput ] = useState< string >( '' );
+	const { getChatInput, saveChatInput } = useChatInputContext();
+	const [ currentInput, setCurrentInput ] = useState( '' );
 	const isOffline = useOffline();
 	const { __ } = useI18n();
 	const lastMessage = messages.length === 0 ? undefined : messages[ messages.length - 1 ];
 	const hasFailedMessage = messages.some( ( msg ) => msg.failedMessage );
 
-	const handleSend = useCallback(
-		async ( messageToSend?: string, isRetry?: boolean ) => {
-			const chatMessage = messageToSend || inputRef.current?.value;
+	// Restore prompt input when site changes
+	useEffect( () => {
+		setCurrentInput( getChatInput( selectedSite.id ) );
+	}, [ selectedSite.id, getChatInput ] );
+
+	// Save prompt input when it changes
+	const setInput = useCallback(
+		( input: string ) => {
+			saveChatInput( input, selectedSite.id );
+			setCurrentInput( input );
+		},
+		[ selectedSite.id, saveChatInput ]
+	);
+
+	const submitPrompt = useCallback(
+		async ( chatMessage: string, isRetry?: boolean ) => {
```

Comment:
Previously, the `handleSend` function accepted passing an undefined value in the `messageToSend` parameter. This was needed because the function was passed to the component `AIInput`, which isn't passing the input value on the handler `handleSend`. Now this function (renamed to `submitPrompt`) is always called with a value, as the chat input is now processed in another function that ensure passing a defined value.